### PR TITLE
Add Apache Tika extension guide

### PR DESCRIPTION
**Pull request by sberyozkin**
@gsmet, I thought I could start with documenting it without the quick start demo available yet and then expand the docs once the demo has been added. Let me know please what you think, I can complete a demo first but it will take a bit longer

### CODE REVIEW

File: docs/src/main/asciidoc/tika-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Using Apache Tika
+
+include::./attributes.adoc[]
+:extension-name: Apache Tika
```
Comment (gsmet) [inline docs/src/main/asciidoc/tika-guide.adoc]: I don't see the purpose of this variable. It makes things less readable and there's no chance we will change it.

And in fact, I don't think you use it.

File: docs/src/main/asciidoc/tika-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    @Consumes({ "application/pdf", "application/vnd.oasis.opendocument.text" })
+    @Produces(MediaType.TEXT_PLAIN)
+    public String extractText(InputStream stream) {
+        return parser.getText(stream);
+    }
+----
+
+== Embedded document text extraction example
+
+package org.acme.tika;
```
Comment (gsmet) [inline docs/src/main/asciidoc/tika-guide.adoc]: You miss the `[source,java]`.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added 2 small comments inline.

I will start the release process in 30 minutes so if you can get them fixed by then, I'll include it :).

File: docs/src/main/asciidoc/tika-guide.adoc (hunk 2 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/tika-guide.adoc]: yeah, just foxed it :-)

File: docs/src/main/asciidoc/tika-guide.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/src/main/asciidoc/tika-guide.adoc]: Ok, will remove it

**Comment by sberyozkin** [comment]
@gsmet thanks, I think I've fixed them :-)

**Comment by gsmet** [comment]
I fixed a few things and amended your commit. Merging now, thanks!

**Review by gsmet — APPROVED** [review]

**Comment by sberyozkin** [comment]
@gsmet thanks for finding the time.

**Comment by sberyozkin** [comment]
@gsmet oh dear, I did not manage to fix the java source mark up from the 1st try :-), keeping fingers crossed I will do it right next time :-)

**Pull request merged**